### ISSUE TITLE
hiopMatrixDense shiftRows segfault

### ISSUE THREAD

**Issue by ashermancinelli** [LLNL/hiop#37]
When testing the `shiftRows` method of `hiopMatrixDense`, I am running into a segfault on our Power9 systems:

```console
233	    A.shiftRows(shift);
(gdb) 

Program received signal SIGSEGV, Segmentation fault.
0x0000000010017474 in hiop::hiopMatrixDense::shiftRows (this=0x7fffffffe410, shift=4) at /people/manc568/projects/hiop/src/LinAlg/hiopMatrix.cpp:256
256	    assert(test1==M[shift<0?0:m_local][0] && "a different copy technique than memcpy is needed on this system");
```
Should we add a fallback method in case memcpy will not work? How do you suggest we test this method? @cnpetra

**Comment by cnpetra** [comment]
test assert test is incorrect for shift > 0

fixed by 06185762879d3e71df7a14de5c71f06f7c71fbc3

**Comment by ashermancinelli** [comment]
After applying your patch I still ran into the a segfault. I implemented the method using `std::rotate` in my branch, would you mind taking a look?

**Comment by cnpetra** [comment]
should be 
`
| 2 2 2 |
| 1 1 1  |
| 2 2 2 |
`
row that are shifted out of the matrix do not enter back at the top of the matrix.

**Comment by ashermancinelli** [comment]
No wrapping, good to know.

**Comment by cnpetra** [comment]
yes, no need to `rotate`

**Comment by cnpetra** [comment]
please confirm whether the master `memcpy` works or shiftrows-forloop-fix `for` copy loop works (aa77c056d28b159f296b93db3d0291aeaf9a8029)?

**Comment by ashermancinelli** [comment]
Your original `memcpy` method is working.
